modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/options/OptionsMediator.java

```diff
@@ -0,0 +1,203 @@
+
+    OptionsResolverFactoryModelProperty optionsProperty = groupModel.getModelProperty(OptionsResolverFactoryModelProperty.class)
+        .orElseThrow(() -> new OptionResolvingException(createStaticMessage(format("The group with name '%s' is not an Option Provider",
+                                                                                   parameterGroupName))));
+
+    try {
+      return resolveOptions(parameterValueResolver, objectSupplier, configurationSupplier, optionsProperty);
+    } catch (Exception e) {
+      throw new RuntimeException("An error occurred trying to fetch options", e);
+    }
+  }
+
+  /**
+   *
+   * @param parameterName
+   * @param parameterValueResolver
+   * @return
+   */
+  public Set<Option> getOptions(String parameterName, ParameterValueResolver parameterValueResolver)
+      throws OptionResolvingException {
+    return getOptions(parameterName, parameterValueResolver, NULL_SUPPLIER, NULL_SUPPLIER);
+  }
+
+  /**
+   *
+   * @param parameterName
+   * @param parameterValueResolver
+   * @param connectionSupplier
+   * @param configurationSupplier
+   * @return
+   */
+  public Set<Option> getOptions(String parameterName, ParameterValueResolver parameterValueResolver,
+                                Supplier<Object> connectionSupplier, Supplier<Object> configurationSupplier)
+      throws OptionResolvingException {
+    ParameterModel parameter = componentModel.getAllParameterModels()
+        .stream()
+        .filter(parameterModel -> parameterModel.getName().equals(parameterName))
+        .findFirst().orElseThrow(() -> new RuntimeException(format("The parameter with name '%s' doesn't exist", parameterName)));
+
+    OptionsResolverFactoryModelProperty modelProperty = parameter.getModelProperty(OptionsResolverFactoryModelProperty.class)
+        .orElseThrow(() -> new RuntimeException(format("The parameter with name '%s' is not an Option Provider", parameterName)));
+
+    try {
+      return resolveOptions(parameterValueResolver, connectionSupplier, configurationSupplier, modelProperty);
+    } catch (Exception e) {
+      throw new RuntimeException(format("An error occurred trying to resolve the Options for parameter '%s' of component '%s'",
+                                        parameterName, componentModel.getName()),
+                                 e);
+    }
+  }
+
+  private Set<Option> resolveOptions(ParameterValueResolver parameterValueResolver, Supplier<Object> connectionSupplier,
+                                     Supplier<Object> configurationSupplier, OptionsResolverFactoryModelProperty optionsProperty)
+      throws NoSuchMethodException, InstantiationException, IllegalAccessException, java.lang.reflect.InvocationTargetException,
+      InitialisationException, ValueResolvingException, OptionResolvingException {
+    Class<? extends OptionsResolver> resolverClass = optionsProperty.getDynamicOptionsResolver();
+
+    OptionsResolver resolver = instantiateClass(resolverClass);
```

Comment:
don't instantiate the resolver each time, copy the metadata case where the factory is stored in a mp and the instances are created once uppon factory cration.